cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -503,25 +503,33 @@ impl PathEnvironment {
         replaced = replaced.replace(FLOX_INSTALL_PLACEHOLDER, packages);
 
         // Replace profile
-        let profile = if let Some(ref custom_profile) = customization.profile {
+        let profile = formatdoc! {r#"
+            # common = """
+            #     fortune | lolcat
+            # """
+            # bash = """
+            #     set -o vi
+            # """
+            # zsh = """
+            #     bindkey -v
+            #     bindkey "^R" history-incremental-search-backward
+            # """"#};
+        replaced = replaced.replace(FLOX_PROFILE_PLACEHOLDER, &profile);
+
+        // Replace the hook
+        let default_hook = if let Some(ref custom_script) = customization.script {
             formatdoc! {r#"
-                common = """
+                on-activate = """
                 {}
-                """"#, indent::indent_all_by(2, custom_profile)}
+                """"#, indent::indent_all_by(2, custom_script)}
         } else {
             formatdoc! {r#"
-                # common = """
-                #   echo "it's gettin flox in here";
+                # on-activate = """
+                #     mkdir my_data_dir
+                #     # Start daemons which set env vars via stdout, e.g. a familiar example:
+                #     eval $(ssh-agent)
```

Comment:
I can think of no more famous example of `eval "$(something ...)"` but then again that's because I remember when `ssh` was the new kid on the block displacing `rsh/rlogin` ... how about `eval "$(starship init bash)"` as a replacement? That's literally quoted on the main page for https://starship.rs/

Reply:
OK ... I did find this [article](https://stackoverflow.com/questions/69488006/i-need-to-re-run-evalssh-agent-and-ssh-add-on-every-boot) which was specifically talking about the challenges of setting up `ssh-agent` but I'm giving up on the search and will hand this over to Ross to find some better examples. Maybe starting a minecraft server or something would be more common these days?!?

Reply:
Are any of these guaranteed to run only once anyway?
I assume if i activate an environment from two different terminals, the on-activate script would run in both no?

Considering that profile scripts are run in different directories, i guess the "clone a git repo with more sources for this environment" case would only work in "on-activate" as `profile` would keep cloning with every subshell.
Generally anything that requires the "original" activation dir (even if its just exporting that for further use in profile script) should go into on-activate is my understanding. -- `eval "$(ssh-agent)"` does indeed feel a bit contrived, and leaning heavily on the activate in bashrc use case.

Other than that I feel we start conflating "what shell is the script run in" with "how often is the script called" now that profile scripts seem to run more often.

Reply:
> I assume if i activate an environment from two different terminals, the on-activate script would run in both no?

Yes - an activation is local to a shell and its child processes, and it can be considered a feature that people can activate distinct instances of the same environment on a host. Conversely, if people want to operate a bunch of processes under the same environment that's why it's so important we be able to replay profile actions in subshells while not repeating the on-activate hook more than once.

> Considering that profile scripts are run in different directories, i guess the "clone a git repo with more sources for this environment" case would only work in "on-activate" as profile would keep cloning with every subshell.

I would imagine that's right. If you're staging data as part of an activation then I imagine that would be to a temporary/ephemeral directory anyway (in order to remain idempotent), and the location of that data would be passed down to subprocesses by way of the environment. In general I'd think that's an unusual use case, but agree that the world will find all kinds of interesting ways to use flox.

> Generally anything that requires the "original" activation dir (even if its just exporting that for further use in profile script) should go into on-activate is my understanding. -- eval "$(ssh-agent)" does indeed feel a bit contrived, and leaning heavily on the activate in bashrc use case.

Yes, as you've noted each subshell can re-source the profile scripts anew, and from a different directory. That is somewhat of a feature in itself as well.

> Other than that I feel we start conflating "what shell is the script run in" with "how often is the script called" now that profile scripts seem to run more often.

Yes the docs will need to be very clear in communicating the role of the various script types and when they run. I think we're getting there, but this will doubtless be an area for improvement over time.

Reply:
I'll meet with Ross this afternoon and come up with a "today" version of the template so we can move forward, understanding that it will undoubtedly be a living bit of text as things evolve over time, and comparably far easier to change than the remainder of this PR. With that I'll go ahead and resolve this conversation and commit to post a followup of our meeting later this afternoon.